engine/installation/linux/ubuntulinux.md

```diff
@@ -116,365 +119,494 @@ packages from the Docker repository:
+Make sure you have satisfied all the
+[prerequisites](ubuntulinux.md#prerequisites), then follow these steps.
 
-1. Log into your Ubuntu installation as a user with `sudo` privileges.
+>**Note**: For production systems, it is recommended that you
+[install a specific version](ubuntulinux.md#install-a-specific-version) so that
+you do not accidentally update Docker. You should plan upgrades for production
+systems carefully.
 
-2. Update your `APT` package index.
+1.  Log into your Ubuntu installation as a user with `sudo` privileges.
 
-        $ sudo apt-get update
+2.  Update your `APT` package index.
 
-3. Install Docker.
+    ```bash
+    $ sudo apt-get update
+    ```
+3.  Install Docker.
 
-        $ sudo apt-get install docker-engine
+    ```bash
+    $ sudo apt-get install docker-engine
+    ```
 
-4. Start the `docker` daemon.
+4.  Start the `docker` daemon.
 
-        $ sudo service docker start
+    ```bash
+    $ sudo service docker start
+    ```
 
-5. Verify `docker` is installed correctly.
+5.  Verify that `docker` is installed correctly by running the `hello-world`
+    image.
 
-        $ sudo docker run hello-world
+    ```bash
+    $ sudo docker run hello-world
+    ```
 
     This command downloads a test image and runs it in a container. When the
-    container runs, it prints an informational message. Then, it exits.
+    container runs, it prints an informational message and exits.
+
+## Install a specific version
+
+To install a specific version of `docker-engine`:
+
+1.  List all available versions using `apt-cache madison`:
+    ```bash
+    $ apt-cache madison docker-engine
+
+    docker-engine | 1.12.3-0~xenial | https://apt.dockerproject.org/repo ubuntu-xenial/main amd64 Packages
+    docker-engine | 1.12.2-0~xenial | https://apt.dockerproject.org/repo ubuntu-xenial/main amd64 Packages
+    docker-engine | 1.12.1-0~xenial | https://apt.dockerproject.org/repo ubuntu-xenial/main amd64 Packages
+    docker-engine | 1.12.0-0~xenial | https://apt.dockerproject.org/repo ubuntu-xenial/main amd64 Packages
+    docker-engine | 1.11.2-0~xenial | https://apt.dockerproject.org/repo ubuntu-xenial/main amd64 Packages
+    docker-engine | 1.11.1-0~xenial | https://apt.dockerproject.org/repo ubuntu-xenial/main amd64 Packages
+    docker-engine | 1.11.0-0~xenial | https://apt.dockerproject.org/repo ubuntu-xenial/main amd64 Packages
+    ```
+
+2.  The second field is the version string. To install exactly `1.12.0-0~xenial`,
+    append it after the package name in the `apt-get install` command, separated
+    from the package name by an equals sign (`=`).
+    ```bash
+    $ sudo apt-get install docker-engine=1.12.0-0~xenial
+    ```
+
+    If you already have a newer version installed, you will be prompted to
+    downgrade Docker. Otherwise, the specific version will be installed.
+
+3.  Follow steps 4 and 5 of
+    [Install the latest version](ubuntulinux.md#install-the-latest-version).
+
+## Install a testing version
```

Comment:
Should we call this "pre-release"?

It may be important to call out that "final" versions are not released through this repository, so to go back to "stable" builds, it's needed to switch back to the "main" repository.

also /cc @andrewhsu @vieux for input (should this be part of the documentation?)